packages/neon_framework/test/request_manager_test.dart

```diff
@@ -0,0 +1,927 @@
+          ]),
+        );
+
+        // ignore: unawaited_futures
+        expectLater(
+          subject.stream,
+          neverEmits([
+            equals(Result.success('VGVzdCB2YWx1ZQ==')),
+            emitsDone,
+          ]),
+        );
+
+        await RequestManager.instance.wrap<String, Uint8List>(
+          'clientID',
+          'key',
+          subject,
+          () => Future.delayed(const Duration(milliseconds: 100), () => utf8.encode('Test value')),
+          (deserialized) => base64Encode(deserialized),
+          (deserialized) => utf8.decode(deserialized),
+          (cached) => utf8.encode(cached),
+          false,
+          false,
+          const Duration(milliseconds: 50),
+        );
+
+        await subject.close();
+      });
+
+      test('throwing request', () async {
+        var subject = BehaviorSubject<Result<String>>();
+
+        // ignore: unawaited_futures
+        expectLater(
+          subject.stream,
+          emitsInOrder([
+            equals(Result<String>.loading()),
+            equals(Result<String>.error('Exception')),
+            emitsDone,
+          ]),
+        );
+
+        await RequestManager.instance.wrap<String, Uint8List>(
+          'clientID',
+          'key',
+          subject,
+          () async => throw Exception(),
+          (deserialized) => base64Encode(deserialized),
+          (deserialized) => utf8.decode(deserialized),
+          (cached) => utf8.encode(cached),
+        );
+
+        await subject.close();
+
+        subject = BehaviorSubject<Result<String>>.seeded(Result.success('Seed value'));
+
+        // ignore: unawaited_futures
+        expectLater(
+          subject.stream,
+          emitsInOrder([
+            equals(Result.success('Seed value')),
+            equals(Result.success('Seed value').asLoading()),
+            equals(Result('Seed value', 'Exception', isLoading: false, isCached: false)),
+            emitsDone,
+          ]),
+        );
+
+        await RequestManager.instance.wrap<String, Uint8List>(
+          'clientID',
+          'key',
+          subject,
+          () async => throw Exception(),
+          (deserialized) => base64Encode(deserialized),
+          (deserialized) => utf8.decode(deserialized),
+          (cached) => utf8.encode(cached),
+        );
+
+        await subject.close();
+      });
+
+      test('throwing DynamiteException should retry', () async {
+        var subject = BehaviorSubject<Result<String>>();
+
+        // ignore: unawaited_futures
+        expectLater(
+          subject.stream,
+          emitsInOrder([
+            equals(Result<String>.loading()),
+            equals(Result<String>.error('DynamiteApiException(statusCode: 500, headers: {}, body: )')),
+            emitsDone,
+          ]),
+        );
+
+        await RequestManager.instance.wrap<String, Uint8List>(
+          'clientID',
+          'key',
+          subject,
+          () async => throw const DynamiteApiException(500, {}, ''),
```

Comment:
You should verify that this method is actually called multiple times.

Reply:
This one and a few other ones are still missing

Reply:
I added an explicit test for this further up.
I like to think that the other tests are to unit test the subjects. If the retry count logic got screwed I want that particular test to fail so I can easily identify the issue and not every test of the request manager.

Reply:
Yeah that makes sense, but the test is called `throwing DynamiteException should retry` so I would expect that it also checks that it correctly retries?

Reply:
you are right. Before I added 97d1b92a3055b598594174be95d4b460e5a93d26 this was obvious as the subject would emit multiple loading events. Now with the global test it makes sense to remove the individual tests as the cache should not play a role in this logic.

Should I remove them (lightening the file) or just add the verifications?

Reply:
Yes, please remove the tests :)